python-backend/tests/mines/now_submissions/resources/test_application_resource.py

```diff
@@ -0,0 +1,118 @@
+
+        assert get_resp.status_code == 200
+        assert len(get_data['records']) == 2
+        assert all(
+            str(application.application_guid) in map(lambda x: x['application_guid'], get_data['records'])
+            for application in [application_1, application_2])
+        assert all(
+            str(application.application_guid) not in map(lambda x: x['application_guid'], get_data['records'])
+            for application in [application_3, application_4])
+
+    def test_get_mine_application_list_filter_by_trackingnumber(self, test_client, db_session, auth_headers):
+        """Should return the records filtered by trackingnumber"""
+
+        mine = MineFactory(minimal=True)
+        application_1 = NOWApplicationFactory(mine=mine, trackingnumber=1)
+        application_2 = NOWApplicationFactory(mine=mine, trackingnumber=1)
+        application_3 = NOWApplicationFactory(mine=mine, trackingnumber=12)
+        application_4 = NOWApplicationFactory(mine=mine, trackingnumber=10305)
+
+        get_resp = test_client.get(f'mines/{mine.mine_guid}/now-submissions/applications?trackingnumber=1',
+                                   headers=auth_headers['full_auth_header'])
+        get_data = json.loads(get_resp.data.decode())
+
+        assert get_resp.status_code == 200
+        assert len(get_data['records']) == 2
+        assert all(
+            str(application.application_guid) in map(lambda x: x['application_guid'], get_data['records'])
+            for application in [application_1, application_2])
+        assert all(
+            str(application.application_guid) not in map(lambda x: x['application_guid'], get_data['records'])
+            for application in [application_3, application_4])
+
+    def test_get_mine_application_list_filter_by_multiple_filters(self, test_client, db_session, auth_headers):
+        """Should return the records filtered by status, noticeofworktype, and tracking email"""
+
+        mine = MineFactory(minimal=True)
+        application_1 = NOWApplicationFactory(mine=mine, status='Rejected', noticeofworktype='dog', trackingnumber=1)
+        application_2 = NOWApplicationFactory(mine=mine, status='Received', noticeofworktype='cat', trackingnumber=1)
+        application_3 = NOWApplicationFactory(mine=mine, status='Rejected', noticeofworktype='dog', trackingnumber=12)
+        application_4 = NOWApplicationFactory(mine=mine, status='Approved', noticeofworktype='cat', trackingnumber=10305)
+
+        get_resp = test_client.get(
+            f'mines/{mine.mine_guid}/now-submissions/applications?status=app,rej&noticeofworktype=dog&trackingnumber=1',
+            headers=auth_headers['full_auth_header'])
+        get_data = json.loads(get_resp.data.decode())
+
+        assert get_resp.status_code == 200
+        assert len(get_data['records']) == 1
+        assert all(
+            str(application.application_guid) in map(lambda x: x['application_guid'], get_data['records'])
```

Comment:
Is this necessary if we're getting a single result?

Reply:
Not necessary, but it's consistent with all of the other tests in this suite, which I think makes it easier to read than it would be if one assertion was structured differently.